client/src/components/App/Content/InvestigationForm/InvestigationForm.tsx

```diff
@@ -81,39 +68,66 @@ const InvestigationForm: React.FC = (): JSX.Element => {
             setSymptomsStartDate, setExposureDate, setHasSymptoms, setEndInvestigationDate]
     );
 
-    const { currentTab, confirmFinishInvestigation, handleSwitchTab, saveCurrentTab, isButtonDisabled, areThereContacts } = useInvestigationForm({ clinicalDetailsVariables, personalInfoData, exposuresAndFlightsVariables, interactedContactsState });
-    
-    const shouldDisableButton = isButtonDisabled(currentTab.name);
-    const isLastTab = (currentTab.id === LAST_TAB_ID || (currentTab.id === LAST_TAB_ID - 1 && !areThereContacts));
-    
+    const { confirmFinishInvestigation, areThereContacts } = useInvestigationForm();
+    const {
+        currentTab,
+        moveToNextTab,
+        setNextTab
+    } = useTabManagement();
+
+    const isInvestigationValid = () => {
+        return !(formsValidations.some((formValidation) => !formValidation));
+    }
+
+    const handleClick = () => {
```

Comment:
handleFormSave clearer

Reply:
this function not saving the form - 
if its the last tab so the finish investigation function called ( not saving the form either)
if its not the last tab the switching tab function is called 
maybe handleNextPageClick / moveToNextTab ?

Reply:
both sound ok

Reply:
changed to handleNextPageClick